### PR TITLE
Add support for big testcase

### PR DESCRIPTION
**Pull request by Bogay**
MongoDB's document has size limit 16MB, use gridfs (`FileField` in `mongoengine`) to solve it.

### CODE REVIEW

**Comment by skps2010** [comment]
依舊沒過

**Comment by Bogay** [comment]
Finish

File: mongo/submission.py [hunk 1]
```diff
@@ -224,19 +224,6 @@ def submit(self, code_file, rejudge=False) -> bool:
                 if submission != self.obj:
                     submission.delete()
 
-        # delete old handwritten submission
```
Comment (skps2010) [inline mongo/submission.py]: 為什麼要拿掉？不是要移除重複的？
這段code的版本是舊的，會導致submission也把自己刪除

File: mongo/submission.py [hunk 2]
```diff
@@ -224,19 +224,6 @@ def submit(self, code_file, rejudge=False) -> bool:
...
-        if self.handwritten:
-            q = {
-                'problem': self.problem,
-                'score': -1,
-                'user': self.user,
-                'handwritten': True
-            }
-
-            for submission in engine.Submission.objects(**q):
-                if submission != self:
```
Comment (skps2010) [inline mongo/submission.py]: 這裡改成 `if submission != self.obj:`
就行了

Comment (skps2010) [reply mongo/submission.py]: 原來是重複的，沒注意到

Comment (Bogay) [reply mongo/submission.py]: BTW, if use `not self == submission` wolud work, maybe

File: mongo/submission.py [hunk 3]
```diff
@@ -140,16 +137,16 @@ def main_code_path(self) -> str:
...
     def get_code(self, path: str) -> str:
-        path = self.code_dir / path
-
-        if not path.exists():
-            raise FileNotFoundError(path)
+        code = self.code_dir / path
+        return code.read_text()
 
-        return path.read_text()
+    def get_comment(self) -> str:
```
Comment (skps2010) [inline mongo/submission.py]: 重複的？

Comment (Bogay) [reply mongo/submission.py]: Which line?

**Pull request merged**